src/lib.rs

```diff
@@ -715,251 +738,307 @@ enum Decoder {
     Match(Expr, Vec<(Pattern, Decoder)>),
     MatchVariant(Expr, Vec<(Pattern, String, Decoder)>),
     Dynamic(DynFormat),
+    Apply(String),
 }
 
 impl Expr {
-    fn eval(&self, scope: &mut Scope) -> Value {
+    fn eval<'a>(&'a self, scope: &'a mut Scope) -> Cow<'a, Value> {
```

Comment:
This comment more applies to the branches of the pattern-match in the method definition, but a lot of explicit `Cow::Owned` can be elided in favor of `.into()` on the `Value`. I believe this can also be done for `Borrowed`, but would have to check.